sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -648,13 +649,80 @@ public override void Dispose()
             }
         }
 
+        private async Task<AmqpSession> CreateSessionIfNeededAsync(AmqpConnection connection, TimeSpan timeout, string transactionGroupKey)
+        {
+            AmqpSession session;
+            FaultTolerantAmqpObject<Controller> faultTolerantController = null;
+            if (transactionGroupKey != null)
+            {
+                if (TransactionGroups.ContainsKey(transactionGroupKey))
+                {
+                    return await TransactionGroups[transactionGroupKey].Session.GetOrCreateAsync(timeout).ConfigureAwait(false);
+                }
+                else
+                {
+                    // each transaction group will have its own session and controller
+                    var faultTolerantSession = new FaultTolerantAmqpObject<AmqpSession>(
+                        async (timeout) =>
+                        {
+                            var stopWatch = ValueStopwatch.StartNew();
+                            AmqpSession session = await CreateAndOpenSessionAsync(connection, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            await CreateControllerAsync(session, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                            return session;
+                        },
+                        async (session) => await session.CloseAsync(timeout).ConfigureAwait(false));
```

Comment:
Should the "Close" delegates here be the safe variant to avoid throwing?

Reply:
hmm.. this wasn't changed from the previous behavior but it is worth looking into.

Reply:
Leaving as is.